app/src/main/java/com/github/se/eduverse/ui/folder/DisplayFolder.kt

```diff
@@ -112,157 +113,152 @@ fun FolderScreen(
+                              },
+                              modifier = Modifier.fillMaxWidth().testTag("sortMode4"),
+                              onClick = {
+                                folderViewModel.sortBy(FilterTypes.ACCESS_RECENT)
+                                sorting = false
+                              })
+                          DropdownMenuItem(
+                              text = { Text("Oldest access", modifier = Modifier.fillMaxWidth()) },
+                              modifier = Modifier.fillMaxWidth().testTag("sortMode5"),
+                              onClick = {
+                                folderViewModel.sortBy(FilterTypes.ACCESS_OLD)
+                                sorting = false
+                              })
+                          DropdownMenuItem(
+                              text = { Text("Most accessed", modifier = Modifier.fillMaxWidth()) },
+                              modifier = Modifier.fillMaxWidth().testTag("sortMode6"),
+                              onClick = {
+                                folderViewModel.sortBy(FilterTypes.ACCESS_MOST)
+                                sorting = false
+                              })
+                          DropdownMenuItem(
+                              text = { Text("Least accessed", modifier = Modifier.fillMaxWidth()) },
+                              modifier = Modifier.fillMaxWidth().testTag("sortMode7"),
+                              onClick = {
+                                folderViewModel.sortBy(FilterTypes.ACCESS_LEAST)
+                                sorting = false
+                              })
+                        }
+                  }
                 }
-              }
 
-          // The files
-          activeFolder!!.files.forEach {
-            Button(
-                onClick = { fileViewModel.openFile(it.fileId, context) },
-                modifier =
-                    Modifier.fillMaxWidth()
-                        .padding(horizontal = 20.dp)
-                        .align(Alignment.Start)
-                        .testTag(it.name)) {
-                  Box(
-                      modifier = Modifier.fillMaxWidth(),
-                      contentAlignment = Alignment.CenterStart) {
-                        Text(it.name)
-                        IconButton(
-                            onClick = {
-                              suppressFile = it
-                              dialogOpen = true
-                            },
-                            modifier =
-                                Modifier.align(Alignment.TopEnd)
-                                    .testTag("delete_icon_${it.name}")) {
-                              Icon(Icons.Default.Close, contentDescription = "Delete File")
-                            }
-                      }
-                }
+            // The files
+            activeFolder!!.files.forEach {
```

Comment:
Using !! to access activeFolder can be risky, as it assumes that this variable will never be null.